nautobot_ssot/tests/test_base_models.py

```diff
@@ -0,0 +1,107 @@
+from typing import Optional, List
+from typing_extensions import TypedDict
+
+from django.contrib.contenttypes.models import ContentType
+from nautobot.extras.models import Tag
+from nautobot.tenancy.models import Tenant, TenantGroup
+from nautobot.utilities.testing import TestCase
+
+from nautobot_ssot.base_models import NautobotModel, BaseAdapter
+
+
+class TagDict(TypedDict):
+    name: str
+
+
+class NautobotTenant(NautobotModel):
+    """A tenant model for testing the `NautobotModel` base class."""
+
+    _model = Tenant
+    _modelname = "tenant"
+    _identifiers = ("name",)
+    _attributes = ("description", "group__name", "tags")
+
+    name: str
+    description: Optional[str] = None
+    group__name: Optional[str] = None
+    tags: List[TagDict] = []
+
+
+class TestAdapter(BaseAdapter):
+    """An adapter for testing the `BaseAdapter` base class."""
+
+    top_level = ("tenant",)
+    tenant = NautobotTenant
```

Comment:
This is the interesting part, as this is all that's needed to have an adapter that does all CRUD operations.

Reply:
it looks clean :)